addon/m3-tracked-array.js

```diff
@@ -25,20 +25,35 @@ export default class M3TrackedArray extends ArrayProxy {
   }
 
   replaceContent(idx, removeAmt, newItems) {
-    newItems = newItems.map((item, idx) => {
+    // Update childModelDatas array
+    // mapping to array of nested models
+    const childModelDatas = this._model._internalModel._modelData.__childModelDatas;
+    const key = this._key;
+    if (childModelDatas && childModelDatas[key] && Array.isArray(childModelDatas[key])) {
+      const newItemsInChildModelData = new Array(newItems.length);
+      Array.prototype.splice.apply(
+        childModelDatas[key],
+        [idx, removeAmt].concat(newItemsInChildModelData)
+      );
+    }
+
+    let startIdx = idx;
+    newItems = newItems.map(item => {
       if (isResolvedValue(item)) {
```

Comment:
I think we should merge this pr without resolving this issue as it's an edge case that we don't actually handle

BUT

Can we please log a bug to remind ourselves to test this case; in particular, to test that inserting a resolved value into a tracked array (and especially at some position other than the start or end) leaves everything in a correct state